crates/ruff_server/src/session.rs

```diff
@@ -220,16 +227,18 @@ impl DocumentSnapshot {
 }
 
 impl Workspaces {
-    fn new(urls: &[Url]) -> crate::Result<Self> {
+    fn new(workspaces: Vec<(Url, ClientSettings)>) -> crate::Result<Self> {
         Ok(Self(
-            urls.iter()
-                .map(Workspace::new)
+            workspaces
+                .into_iter()
+                .map(|(url, settings)| Workspace::new(&url, settings))
                 .collect::<crate::Result<_>>()?,
         ))
     }
 
     fn open_workspace_folder(&mut self, folder_url: &Url) -> crate::Result<()> {
-        let (path, workspace) = Workspace::new(folder_url)?;
+        // TODO(jane): find a way to allow for workspace settings to be updated dynamically
```

Comment:
Nit: can we create an issue to track this that outlines the problem (and possible options)?

Reply:
Issue created here: https://github.com/astral-sh/ruff/issues/10797